core/src/main/java/org/hisp/dhis/android/core/trackedentity/search/TrackedEntitySearchItemHelper.kt

```diff
@@ -50,7 +50,7 @@ object TrackedEntitySearchItemHelper {
             syncState = i.syncState(),
             aggregatedSyncState = i.aggregatedSyncState(),
             deleted = i.deleted() ?: false,
-            isOnline = i.id() == null,
+            isOnline = i.syncState() == null,
```

Comment:
Checked that it is safe to use this property to tell apart the online instances.